samples/nextjs/src/pages/sitecore-page.test.tsx

```diff
@@ -0,0 +1,90 @@
+import 'jsdom-global/register';
+import { expect, use } from 'chai';
+import sinonChai from 'sinon-chai';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+import { SitecorePageProps } from 'lib/page-props';
+import SitecorePage from './[[...path]]';
+import { layoutPersonalizationService } from 'lib/layout-personalization-service';
+
+use(sinonChai);
+
+describe('SitecorePage', () => {
+  describe('useMemo', () => {
+    beforeEach(function () {
+      sinon.spy(layoutPersonalizationService, 'loadPersonalization');
+    });
+
+    afterEach(function () {
+      sinon.restore();
+    });
+
+    it('should call loadPersonalization once with passed props', () => {
+      shallow(<SitecorePage {...sitecorePageProps} />);
+      expect(layoutPersonalizationService.loadPersonalization).to.have.been.callCount(1);
+      expect(layoutPersonalizationService.loadPersonalization).to.have.been.calledWith(
+        {
+          route: sitecorePageProps?.layoutData?.sitecore?.route,
+          itemId: sitecorePageProps?.layoutData?.sitecore?.route?.itemId,
+          ...sitecorePageProps?.layoutData?.sitecore?.context,
+        },
+        sitecorePageProps?.layoutData?.sitecore?.route
+      );
+    });
+
+    it('should return empty component rendering if application is running in disconnected mode', () => {
```

Comment:
The tests don't check the return result. They check if necessary calls were made. I'd rename them accordingly. For instance: `should return empty component rendering if application is running in disconnected mode` => `should not load personalization if application is running in disconnected mode`. Actual for all tests.

Reply:
This test feature will be totally refactored